### PR TITLE
fix: respect order_tax_mode

### PR DESCRIPTION
**Pull request by dominik-stripe**
This PR ensure that our connector takes an order's `order_tax_mode` into account.

### CODE REVIEW

File: src/Entity/MiraklOrder.php [hunk 1]
```diff
@@ -30,6 +30,7 @@ abstract public function getCustomerId(): string;
     abstract public function getTransactionNumber(): string;
 
     abstract public function isValidated(): bool;
+    abstract public function taxIncluded(): bool;
```
Comment (youssef-stripe) [inline src/Entity/MiraklOrder.php:33]:
```suggestion
    abstract public function isTaxIncluded(): bool;
```

File: src/Entity/MiraklProductOrder.php [hunk 2]
```diff
@@ -49,6 +49,11 @@ public function isValidated(): bool
         ]);
     }
 
+    public function taxIncluded(): bool
```
Comment (youssef-stripe) [inline src/Entity/MiraklProductOrder.php:52]:
```suggestion
    public function isTaxIncluded(): bool
```

File: src/Entity/MiraklProductOrder.php [hunk 3]
```diff
@@ -62,6 +67,12 @@ public function isAborted(): bool
     public function getAmountDue(): float
     {
         $amount = $this->order['total_price']; // REFUSED/CANCELED are already not included
+
+        // Don't add taxes to the total_price if taxes are included
+        if ($this->taxIncluded()) {
```
Comment (youssef-stripe) [inline src/Entity/MiraklProductOrder.php:72]:
```suggestion
        if ($this-> isTaxIncluded()) {
```

File: src/Entity/MiraklProductOrder.php [hunk 4]
```diff
@@ -78,7 +89,10 @@ public function getAbortedAmount(): float
             switch ($orderLine['order_line_state']) {
                 case 'REFUSED':
                     $amount += (float) $orderLine['total_price'];
-                    $amount += (float) $this->getOrderLineTaxes($orderLine);
+
+                    if (!$this->taxIncluded()) {
```
Comment (youssef-stripe) [inline src/Entity/MiraklProductOrder.php:93]:
```suggestion
                    if (!$this->isTaxIncluded()) {
```

File: src/Entity/MiraklServiceOrder.php [hunk 5]
```diff
@@ -49,6 +49,11 @@ public function isValidated(): bool
         ]);
     }
 
+    public function taxIncluded(): bool
```
Comment (youssef-stripe) [inline src/Entity/MiraklServiceOrder.php:52]:
```suggestion
    public function isTaxIncluded(): bool
```

File: src/Entity/MiraklServiceOrder.php [hunk 6]
```diff
@@ -62,8 +67,11 @@ public function isAborted(): bool
     public function getAmountDue(): float
     {
         $taxes = 0;
-        foreach (($this->order['price']['taxes'] ?? []) as $tax) {
-            $taxes += (float) $tax['amount'];
+
+        if (!$this->taxIncluded()) {
```
Comment (youssef-stripe) [inline src/Entity/MiraklServiceOrder.php:71]:
```suggestion
        if (!$this->isTaxIncluded()) {
```

**Review by youssef-stripe — CHANGES_REQUESTED** [review]
Looks good to me! Just a minor change to the function name to use the same format as other functions returning a boolean.

**Comment by dominik-stripe** [comment]
Done!

**Review by youssef-stripe — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Version 3.0.5
+
+- The connector now takes Mirakl's `order_tax_mode` into account when calculating transfer amounts.
```
Comment (youssef-stripe) [inline CHANGELOG.md:5]:
```suggestion
- Fix: take Mirakl's `order_tax_mode` into account when calculating transfer amounts.
```

**Pull request merged**